### PR TITLE
Change GIF max matrix size error to explicitly mention GIF files

### PR DESCRIPTION
**Pull request by ClearlyClaire**

### CODE REVIEW

File: app/models/concerns/attachmentable.rb [hunk 1]
```diff
@@ -52,9 +52,12 @@ def check_image_dimension(attachment)
     return if attachment.blank? || !/image.*/.match?(attachment.content_type) || attachment.queued_for_write[:original].blank?
 
     width, height = FastImage.size(attachment.queued_for_write[:original].path)
-    matrix_limit  = attachment.content_type == 'image/gif' ? GIF_MATRIX_LIMIT : MAX_MATRIX_LIMIT
 
-    raise Mastodon::DimensionsValidationError, "#{width}x#{height} images are not supported" if width.present? && height.present? && (width * height > matrix_limit)
+    if attachment.content_type == 'image/gif'
+      raise Mastodon::DimensionsValidationError, "#{width}x#{height} GIF files are not supported" if width.present? && height.present? && (width * height > GIF_MATRIX_LIMIT)
+    elsif width.present? && height.present? && (width * height > MAX_MATRIX_LIMIT)
```
Comment (mjankowski) [inline app/models/concerns/attachmentable.rb:58]: Thoughts on keeping the `if width.present? && height.present?` check outside the `content_type` conditional? Might keep the check about the error raise more narrowly focused on what's different based on type (the matrix comparison).

**Review by mjankowski — APPROVED** [review]
One nuanced comment, but looks fine in general.

**Comment by vmstan** [comment]
Potentially closes #26099

**Comment by ClearlyClaire** [comment]
> Potentially closes #26099

I'm not sure. The error message wouldn't be “Can't create thumbnail”. That being said, trying to replicate that issue right now doesn't end up with “Can't create thumbnail” but with “970x990 images are not supported”, which would be changed by the current PR.

**Review by mjankowski — APPROVED** [review]

**Review by renchap — APPROVED** [review]

**Pull request merged**